### ISSUE TITLE
/etc/hostname workaround

### ISSUE THREAD

**Issue by leshik** [krebs/krops#27]
Hi,

Is it possible to work around the `/etc/hostname` file and the `HOSTNAME` env variable [here](https://github.com/krebs/krops/blob/master/lib/default.nix#L54) somehow? I'm deploying from Nix on macOS to the Linux VPS; macOS doesn't have `/etc/hostname`, and while `HOSTNAME` is set in the shell, it's not exported. I can work around right now by setting a dummy `HOSTNAME`, e.g. `$(HOSTNAME=localhost nix-build --no-out-link krops.nix)`, but it would be nice if it were working out of the box.

**Comment by 4z3** [comment]
Yes, the `/etc/hostname` file and the `HOSTNAME` variable are only used to set the default value.  Neither of those should be required when explicitly specifying a host in `writeDeploy`'s `target` attribute, e.g. `target = "localhost";`. See https://github.com/krebs/krops#target for details.

**Comment by leshik** [comment]
@4z3 yes, it works this way, but I need to use mkTarget as I want to specify `sudo = true`. Then it doesn't work.

**Comment by 4z3** [comment]
Doesn't this work?:

```nix
  target = lib.mkTarget "localhost" // {
    sudo = true;
  };
```

**Comment by leshik** [comment]
@4z3 no, it doesn't:
```
error: opening file '/etc/hostname': No such file or directory
(use '--show-trace' to show detailed location information)
```

**Comment by 4z3** [comment]
Can you paste/link your krops configuration? I've deleted `/etc/hostname` locally and `unset HOSTNAME`, but I cannot reproduce the issue with following snippet:

```nix
pkgs.krops.writeDeploy "deploy" {
    source = { /* ... */ };
    target = lib.mkTarget "root@localhost" // {
      extraOptions = [
        "-oLogLevel=DEBUG3"
      ];
    };
  }
```

**Comment by 4z3** [comment]
OK, I could reproduce it by using `sudo = true;`.

**Comment by 4z3** [comment]
This issue has been addressed in https://github.com/krebs/krops/commit/c2fa48550f2bb46009b9cecdb9ac838dc402ce19, released as [v1.24.1](https://github.com/krebs/krops/tree/v1.24.1). @leshik can you check if this fix works for you?

**Comment by leshik** [comment]
@4z3 yeah, that works perfectly.
Thank you for the quick fix!
